### ISSUE TITLE
[RELEASE] Alpha13

### ISSUE THREAD

**Issue by tdesveauxPKFX** [premake/premake-core#1173]
Let's talk about the alpha 13 release.

There is a candidate branch (release-5.0-alpha13), this issue is to gather feedback on it.

- [x] Update copyright to 2018
- [x] Add iOS to `os` options
- [x] Sort `--help` lists
- [x] Rewrite `PRlogs_to_RELEASE_format.py` in lua

- [x] Build and test binaries on Desktops
    - [x] Windows
    - [x] OSX
    - [x] Linux

First feedback from @samsinsane (in #1026):
> 
> 
> Notes/Feedback:
> 
>     * The branch appears to contain all commits from master
> 
>     * `premake --version` looks correct
> 
>     * `premake --help` looks correct except for the copyright ending at 2017
>       
>       * Perhaps we should have support for iOS in the `os` option?
>       * The `android` value for the `os` option is at the end of the list due to how it's added, perhaps all the lists should be sorted before being displayed by `--help`?
> 
>     * CHANGES.txt looks correct.
> 
>     * `PRlogs_to_RELEASE_format.py` could be written in Lua instead, similar to `package.lua`. It could even extend `package.lua`?
> 
> 
> Unrelated:
> 
>     * The Premake project isn't set to be the `startproject`, which is kind of strange
> 
>     * The CHANGES.txt reads very weird because it never mentions the current release by name. It says "Since 5.0-alpha12:" instead of "Changes in 5.0-alpha13:" which I think would make slightly more sense?
> 
> 
> Nothing else jumps out at me right now, but @starkos might have some additional notes about the whole process. Also, when generating the Linux builds, try to use a distro that has glibc 2.16 so that we don't exacerbate the issue detailed in #632. Perhaps use the same distro as @starkos, whatever that happens to be?

**Comment by tdesveauxPKFX** [comment]
>       * Perhaps we should have support for iOS in the `os` option?

For iOS, if it works without changes from the user, yes we should add it.

>       * The `android` value for the `os` option is at the end of the list due to how it's added, perhaps all the lists should be sorted before being displayed by `--help`?

That would be nicer.

>     * `PRlogs_to_RELEASE_format.py` could be written in Lua instead, similar to `package.lua`. It could even extend `package.lua`?

Sure, it was mainly a cp from an other script of mine, it make more sense to have it in lua.
Not sure about adding it to `package.lua` yet. If you run it, you will see some really old PRs are listed, I need to look into why but I think we should do 4-5 release with it and automate the process if everything goes well.

>     * The Premake project isn't set to be the `startproject`, which is kind of strange

Strange indeed, it was always set correctly as far I remember.

> 
>     * The CHANGES.txt reads very weird because it never mentions the current release by name. It says "Since 5.0-alpha12:" instead of "Changes in 5.0-alpha13:" which I think would make slightly more sense?

I agree, it took me a second to understand that the alpha12 changes didn't disappear.

> Nothing else jumps out at me right now, but @starkos might have some additional notes about the whole process. Also, when generating the Linux builds, try to use a distro that has glibc 2.16 so that we don't exacerbate the issue detailed in #632. Perhaps use the same distro as @starkos, whatever that happens to be?

According to [this](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Linux_compatibility_matrix), I'd say building with GCC4 (glibc 2.17) should be fine.

**Comment by samsinsane** [comment]
> If you run it, you will see some really old PRs are listed, I need to look into why

If they're the ones I saw, they're part of the Android module. I maintained the history when I added the module into the repo, which includes all of the merge commits. I don't have Python3 installed locally, but it looks like using `--first-parent` omits the "bad" merge commits. Can you try changing `git_command` to have `--first-parent` as well as `--merges` and see if that produces the same list of PRs you had before?

> I think we should do 4-5 release with it and automate the process if everything goes well.

Sounds good to me! :+1:

> Strange indeed, it was always set correctly as far I remember.

I think we used to set the first project specified as the start up project, but that seems to be gone now? I would normally suggest that I did something wrong, but it happened after a full clean and bootstrap run. Very odd.

> According to this, I'd say building with GCC4 (glibc 2.17) should be fine.

That is an amazing source, I wish I knew this existed before! We could also consider using something like [musl](https://www.musl-libc.org/) which I'm fairly certain I've seen suggested as a solution to the glibc issue.

**Comment by tdesveauxPKFX** [comment]
> If they're the ones I saw, they're part of the Android module. I maintained the history when I added the module into the repo, which includes all of the merge commits. I don't have Python3 installed locally, but it looks like using --first-parent omits the "bad" merge commits. Can you try changing git_command to have --first-parent as well as --merges and see if that produces the same list of PRs you had before?

Fixed the issue, thanks! I will add it when rewriting in lua.

> That is an amazing source, I wish I knew this existed before! We could also consider using something like musl which I'm fairly certain I've seen suggested as a solution to the glibc issue.

Yeah, really useful (found it when I linked it, would have been so useful to me a month ago...).
As for musl, we could try it but it seems it is slower than glibc in quite a few points. It wins only on binary size which is less a concern than speed I think.
We should benchmark before making the switch.

We could also use symver but building against gcc-4 is less painful IMO.

**Comment by samsinsane** [comment]
Another thing to fix up, in the `README.md`, the last badge needs to be updated to point at alpha13 instead of alpha12. Unfortunately, you won't be able to verify that you've done it correctly until the release/tag exists.

**Comment by tdesveauxPKFX** [comment]
Well, I wasn't able to test the release yet... I should be able tomorrow.

I planned to update the tag as a last thing (just before merging release back to master), should I do it sooner?

**Comment by tdesveauxPKFX** [comment]
Here we go! Just published the release, everything should be fine (hopefully).

Just need to update the site and post on the forums now (which ones?).

**Comment by samsinsane** [comment]
The text at the top said "alpha12" in relation to the source package, I've fixed this up. Great work @tdesveauxPKFX! I'm excited to see feedback on this latest release.

The forums should be the Google group, I believe?

Also, still need to update the badge in the README:
![image](https://user-images.githubusercontent.com/11887457/47192225-52ef5200-d38f-11e8-9406-777543be759c.png)
and I think submit a PR for the alpha13 release into master? Or maybe the branch was merged locally and just pushed to master?

**Comment by tdesveauxPKFX** [comment]
> The text at the top said "alpha12" in relation to the source package, I've fixed this up. Great work @tdesveauxPKFX! I'm excited to see feedback on this latest release.

Thanks! (I obviously had to miss something)

> The forums should be the Google group, I believe?

Really have no idea. I hope @starkos can enlighten us on that point (as well as post on twitter).

> Also, still need to update the badge in the README:
> ![image](https://user-images.githubusercontent.com/11887457/47192225-52ef5200-d38f-11e8-9406-777543be759c.png)

I changed it in the `release` branch after merging the `release-5.0-alpha13` branch in it since it is the one I tagged.

> and I think submit a PR for the alpha13 release into master? Or maybe the branch was merged locally and just pushed to master?

Will do it now, kinda had to rush out after posting the release.

A last note about releases. I think we should plan the next one for early December (ideally first week) then try to plan a monthly release from there. Thoughts?
There is some points I found painful with the release process and will try work on them before the next one (For example, I had to manually package the windows binary as the package script does not work on Windows). I may also look into automating the full release process if I find the time.
